modules/hal_sensors/i2c_sensor_driver/src/I2C_sensor_driver.cpp

```diff
@@ -36,80 +36,84 @@ constexpr uint8_t GetLowerByte(uint16_t number) {
   return (number & 0xFF);
 }
 
-void I2CDriver::Init() {
-  i2c_peripheral_->begin();
+void I2C_sensor_driver::init_i2c_helper(void) {
+  hal::i2c::I2C_Driver::begin();
 }
 
-void I2CDriver::ChangeAddress(uint8_t new_i2c_address) {
-  i2c_addr_ = new_i2c_address;
+void I2C_sensor_driver::ChangeAddress(hal::i2c::I2CAddr new_i2c_address) {
+  this->set_i2c_addr(new_i2c_address);
 }
 
-void I2CDriver::WriteReg(uint16_t reg, uint8_t data) {
+void I2C_sensor_driver::write8_reg16b(uint16_t reg, uint8_t data) {
   const uint8_t kRegLowerByte = GetLowerByte(reg);
   const uint8_t kRegUpperByte = GetUpperByte(reg);
 
-  i2c_peripheral_->beginTransmission(i2c_addr_);
-  i2c_peripheral_->write(kRegUpperByte);
-  i2c_peripheral_->write(kRegLowerByte);
-  i2c_peripheral_->write(data);
-  i2c_peripheral_->endTransmission();
+  this->beginTransmission(this->get_i2c_addr());
+  this->write(kRegUpperByte);
+  this->write(kRegLowerByte);
+  this->write(data);
+  this->endTransmission();
 }
 
-void I2CDriver::WriteReg16(uint16_t reg, uint16_t data) {
+void I2C_sensor_driver::write16_reg16b(uint16_t reg, uint16_t data) {
   const uint8_t kRegLowerByte = GetLowerByte(reg);
   const uint8_t kRegUpperByte = GetUpperByte(reg);
 
-  i2c_peripheral_->beginTransmission(i2c_addr_);
-  i2c_peripheral_->write(kRegUpperByte);
-  i2c_peripheral_->write(kRegLowerByte);
+  this->beginTransmission(this->get_i2c_addr());
+  this->write(kRegUpperByte);
+  this->write(kRegLowerByte);
 
   uint8_t temp;
   temp = GetUpperByte(data);
-  i2c_peripheral_->write(temp);
+  this->write(temp);
   temp = GetLowerByte(data);
-  i2c_peripheral_->write(temp);
-  i2c_peripheral_->endTransmission();
+  this->write(temp);
+  this->endTransmission(true);
 }
 
-uint8_t I2CDriver::ReadReg(uint16_t reg) {
+uint8_t I2C_sensor_driver::send_read8_reg16b(uint16_t reg) {
   const uint8_t kRegLowerByte = GetLowerByte(reg);
   const uint8_t kRegUpperByte = GetUpperByte(reg);
 
-  i2c_peripheral_->beginTransmission(i2c_addr_);
-  i2c_peripheral_->write(GetUpperByte(kRegUpperByte));
-  i2c_peripheral_->write(GetLowerByte(kRegLowerByte));
-  i2c_peripheral_->endTransmission(false);
-  i2c_peripheral_->requestFrom(i2c_addr_, 1);
-  uint8_t data = i2c_peripheral_->read();
+  this->beginTransmission(this->get_i2c_addr());
+  this->write(GetUpperByte(kRegUpperByte));
+  this->write(GetLowerByte(kRegLowerByte));
+  this->endTransmission(false);
+  this->requestFrom(this->get_i2c_addr(), 1, true);
+  uint8_t data = this->read();
   return data;
 }
 
-uint16_t I2CDriver::ReadReg16(uint16_t reg) {
+uint16_t I2C_sensor_driver::send_read16_reg16(uint16_t reg) {
```

Comment:
The naming scheme of functions became even weirder... Now you read 16-bits out of 16bit reg... its logical that you read 16-bits out of 16-bits reg right?. So don't have to specify that you read 16-bits twice in the name.. Previous name was obviously better

Reply:
And why is send in the name... wtf